### PR TITLE
[MRG] DOC: start example for workflow

### PR DESCRIPTION
**Pull request by jasmainak**
supersedes #205

### CODE REVIEW

**Comment by jasmainak** [comment]
okay, got stuck at step 1. cc @hoechenberger @alexrockhill I get this:

```py
In [2]: %run examples/plot_autoreject_workflow.py                               

👋 Hello! This is openneuro-py 2021.2. Great to see you! 🤗

   👉 Please report problems 🤯 and bugs 🪲 at
      https://github.com/hoechenberger/openneuro-py/issues

🌍 Preparing to download ds002778 …
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
~/Documents/github_repos/autoreject/examples/plot_autoreject_workflow.py in <module>
     34 target_dir = mne.utils._TempDir()
     35 openneuro.download(dataset=dataset, target_dir=target_dir,
---> 36                    include=[f'sub-{subject_id}/'])
     37 
     38 # %%

~/anaconda3/lib/python3.7/site-packages/openneuro/download.py in download(dataset, tag, target_dir, include, exclude, verify_hash, verify_size, max_retries, max_concurrent_downloads)
    416                                       tag=tag,
    417                                       max_retries=max_retries,
--> 418                                       retry_backoff=retry_backoff)
    419 
    420     files = []

~/anaconda3/lib/python3.7/site-packages/openneuro/download.py in _get_download_metadata(base_url, dataset_id, tag, max_retries, retry_backoff)
     74                                       retry_backoff=retry_backoff)
     75     else:
---> 76         raise RuntimeError(f'Error {response.status_code} when trying to '
     77                            f'fetch metadata.')
     78 

RuntimeError: Error 404 when trying to fetch metadata.
```

**Comment by hoechenberger** [comment]
@jasmainak This is an outdated version of openneuro-py, can you try updating to the latest version (2021.8)?

**Comment by jasmainak** [comment]
sorry about the false alarm, works now!

**Comment by jasmainak** [comment]
alright, I massaged the tutorial a bit that was started by @alexrockhill 

good to go from my end @agramfort @dengemann 

https://869-59485140-gh.circle-artifacts.com/0/html/auto_examples/plot_autoreject_workflow.html#sphx-glr-auto-examples-plot-autoreject-workflow-py

**Comment by hoechenberger** [comment]
Sorry to crash the party but when working on the MNE BIDS Pipeline, we recently found the global rejection thresholds for pre-ICA rejection not usable for EEG data with electrodes placed closely to the eyes. The captured EOG activity would lead to the rejection of large quantities of epochs. So we ended up NOT using autoreject for pre-ICA rejection and only use it after EOG artifacts have been removed (i.e., after ICA cleaning)

**Comment by jasmainak** [comment]
Humm I see. What threshold did you end up using? Would you recommend the same for MEG data?

Also, I had a recent discussion about this -- EEGLAB allows you to reject epochs based on groups of channels that are not channel types. For example, you should be able to do:

```py
from mne import pick_types

picks = pick_types(epochs.info, eeg=True, exclude=['Fp1', 'Fp2'])
ica = ICA(..., reject=dict(picks=200e-6))
```

This is the most optimal for the ICA algorithm. I had some users asking about this kind of functionality and I ended up asking them to code it manually.

**Comment by jasmainak** [comment]
@agramfort @dengemann any comments from your ends? @hoechenberger any further thoughts?

File: examples/plot_autoreject_workflow.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+# %%
+# visualize the dropped epochs
+epochs[reject_log.bad_epochs].plot(scalings=dict(eeg=100e-6))
+
+# %%
+# and the reject log
+reject_log.plot('horizontal')
+
+# %%
+# Autoreject with Highpass filter
```
Comment (agramfort) [inline examples/plot_autoreject_workflow.py:92]:
```suggestion
# Autoreject with high-pass filter
```
according to wikipedia

File: examples/plot_autoreject_workflow.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+# -------------------------------
+# The data may be very valuable and the time for the experiment
+# limited and so we may want to take steps to reduce the number of
+# epochs dropped by first using other steps to preprocess the data.
+# We will use a highpass filter first to remove slow drift that could
```
Comment (agramfort) [inline examples/plot_autoreject_workflow.py:97]:
```suggestion
# We will use a high-pass filter first to remove slow drift that could
```

File: examples/plot_autoreject_workflow.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+                           n_jobs=1, verbose='tqdm')
+ar.fit(epochs[:20])  # fit on a few epochs to save time
+epochs_ar, reject_log = ar.transform(epochs, return_log=True)
+
+# %%
+# visualize the dropped epochs
+epochs[reject_log.bad_epochs].plot(scalings=dict(eeg=100e-6))
+
+# %%
+# and the reject log. As we can see in the plot, highpass filtering reduced
```
Comment (agramfort) [inline examples/plot_autoreject_workflow.py:119]:
```suggestion
# and the reject log. As we can see in the plot, high-pass filtering reduced
```
to update everywhere

**Review by agramfort — APPROVED** [review]
just nitpicks

LGTM

**Comment by jasmainak** [comment]
okay I'm done here @agramfort 

pushed a cosmit here: https://github.com/autoreject/autoreject/pull/221/commits/072f8cfbe3b4277d0ca2a98b46fdfefc4bc0438e

Basically recommending local autoreject pre-ICA since @hoechenberger reported global autoreject drops too many epochs

Feel free to merge if you're happy

**Comment by agramfort** [comment]
thx @jasmainak

**Comment by hoechenberger** [comment]
Thank you, @jasmainak! I wasn't actually "aware" that autoreject local before ICA might be an option. But I think it's a great idea!

**Comment by dengemann** [comment]
I have always done AR local before ICA as far as I remember. It really gave better results for me.

**Pull request merged**